### PR TITLE
Add unit tests to cover scheduler's setup

### PR DESCRIPTION
**Pull request by Huang-Wei**
#### What type of PR is this?

/kind cleanup
/sig scheduling

#### What this PR does / why we need it:

Follow-up of https://github.com/kubernetes/kubernetes/pull/105915#issuecomment-958576747.

This PR adds unit tests to prove #105915's accuracy. 

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: cmd/kube-scheduler/app/server_test.go [hunk 1]
```diff
@@ -277,13 +373,25 @@ profiles:
...
-				gotPlugins[n] = p.ListPlugins()
+			if tc.wantPlugins != nil {
+				gotPlugins := make(map[string]*config.Plugins)
+				for n, p := range sched.Profiles {
+					gotPlugins[n] = p.ListPlugins()
+				}
+
+				if diff := cmp.Diff(tc.wantPlugins, gotPlugins); diff != "" {
+					if diff := cmp.Diff(tc.wantPlugins, gotPlugins); diff != "" {
+						t.Errorf("Unexpected plugins diff (-want, +got): %s", diff)
```
Comment (kerthcet) [inline cmd/kube-scheduler/app/server_test.go:384]: duplicate code

File: cmd/kube-scheduler/app/server_test.go [hunk 2]
```diff
@@ -277,13 +373,25 @@ profiles:
...
-			gotPlugins := make(map[string]*config.Plugins)
-			for n, p := range sched.Profiles {
-				gotPlugins[n] = p.ListPlugins()
+			if tc.wantPlugins != nil {
+				gotPlugins := make(map[string]*config.Plugins)
+				for n, p := range sched.Profiles {
+					gotPlugins[n] = p.ListPlugins()
+				}
+
+				if diff := cmp.Diff(tc.wantPlugins, gotPlugins); diff != "" {
```
Comment (kerthcet) [inline cmd/kube-scheduler/app/server_test.go:382]: duplicate conditional statement?

File: cmd/kube-scheduler/app/server_test.go (hunk 1 above)
Comment (Huang-Wei) [reply cmd/kube-scheduler/app/server_test.go:384]: oops. Fixed.

File: cmd/kube-scheduler/app/server_test.go (hunk 2 above)
Comment (alculquicondor) [reply cmd/kube-scheduler/app/server_test.go:382]: what do you mean?

Comment (kerthcet) [reply cmd/kube-scheduler/app/server_test.go:382]: it's already fixed, forget it plz.

**Review by alculquicondor — COMMENTED** [review]
/lgtm

**Pull request merged**